### PR TITLE
Ppd 255 donation service integration

### PR DESCRIPTION
**Pull request by alessio-acitelli**
#### List of Changes

- connected to GPS the external service

#### Motivation and Context

- allow the creation of a spontaneous payment

#### How Has This Been Tested?

Manual test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/gov/pagopa/spontaneouspayment/config/FeignConfig.java [hunk 1]
```diff
@@ -12,4 +13,9 @@ public class FeignConfig {
     RequestInterceptor requestIdInterceptor() {
         return requestTemplate -> requestTemplate.header(HEADER_REQUEST_ID, MDC.get("requestId"));
     }
+    
+    @Bean
+    Logger.Level feignLoggerLevel() {
+        return Logger.Level.FULL;
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/spontaneouspayment/config/FeignConfig.java]: is it needed?

File: src/test/java/it/gov/pagopa/spontaneouspayment/service/PaymentsServiceTest.java [hunk 2]
```diff
@@ -208,11 +212,30 @@ void createSpontaneousPayment() throws IOException {
 		// precondition
 		PaymentPositionModel paymentModel = MockUtil.readModelFromFile("gpd/getPaymentPosition.json",
 				PaymentPositionModel.class);
+		String extServicePOString = "{\r\n"
```
Comment (jacopocarlini) [inline src/test/java/it/gov/pagopa/spontaneouspayment/service/PaymentsServiceTest.java]: for a more elegant code, read the json from a file

File: src/test/resources/application.properties [hunk 3]
```diff
@@ -5,8 +5,8 @@ info.application.description=@project.description@
 info.properties.environment=JUNIT
 
 # Cosmos account config
-azure.cosmos.uri=https://pagopa-d-gps-cosmos-account.documents.azure.com:443/
-azure.cosmos.key=lGiI05RuRfjbT3UvHzFjuITKcSu6gK0f8lgHdKmipHgGiTIKVxceTdpJhVC67xmYm5uXihdsznnyyfOm2LsVoQ==
+azure.cosmos.uri=https://pagopa-d-weu-gps-cosmos-account.documents.azure.com:443/
+azure.cosmos.key=f7162UkE9ZixAZZ4ln86dzZOQ2Im68Q6IYah56MpsBTpuydDo3Btd3T5hmlPAgq9jkGIfD43TnUhm6COokcVTA==
```
Comment (jacopocarlini) [inline src/test/resources/application.properties:9]: ⚠️

File: src/main/java/it/gov/pagopa/spontaneouspayment/service/ExternalServiceClient.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import it.gov.pagopa.spontaneouspayment.config.FeignConfig;
+
+
+@FeignClient(value = "externalservice", url = "runtime_url", configuration = FeignConfig.class)
+public interface ExternalServiceClient {
+
+    @Retryable(exclude = FeignException.FeignClientException.class, maxAttemptsExpression = "${retry.ext-service.maxAttempts}",
+            backoff = @Backoff(delayExpression = "${retry.ext-service.maxDelay}"))
+    @PostMapping(value = "/paymentoptions", produces=MediaType.APPLICATION_JSON_VALUE, consumes=MediaType.APPLICATION_JSON_VALUE)
+    String getPaymentOption(URI baseUrl,
```
Comment (jacopocarlini) [inline src/main/java/it/gov/pagopa/spontaneouspayment/service/ExternalServiceClient.java:22]: why map the response in a string?
You can use `PaymentOptionModel` as result and delete the `ConvertJsonPOToPaymentOptionModel` class

**Review by jacopocarlini — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/spontaneouspayment/config/FeignConfig.java (hunk 1 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/spontaneouspayment/config/FeignConfig.java]: Ops sorry, it's a typo i delete it

File: src/test/java/it/gov/pagopa/spontaneouspayment/service/PaymentsServiceTest.java (hunk 2 above)
Comment (alessio-acitelli) [reply src/test/java/it/gov/pagopa/spontaneouspayment/service/PaymentsServiceTest.java]: An elegant choice only for a string in a test? :)
Ok :)

File: src/main/java/it/gov/pagopa/spontaneouspayment/service/ExternalServiceClient.java (hunk 4 above)
Comment (alessio-acitelli) [reply src/main/java/it/gov/pagopa/spontaneouspayment/service/ExternalServiceClient.java:22]: I tried but doesn't work, the only way i found was by string

**Review by jacopocarlini — APPROVED** [review]

**Pull request merged**